src/session.ts

```diff
@@ -96,41 +97,72 @@ async function withAuth(request: Request, { ensureSignedIn = false, debug = fals
         },
       });
     }
-    return { user: null };
+
+    return json(
+      {
+        user: null,
+        ...data,
+      },
+      {
+        headers,
+      },
+    );
   }
 
-  const { sid: sessionId, org_id: organizationId, role, permissions } = decodeJwt<AccessToken>(session.accessToken);
+  const { sessionId, organizationId, role, permissions } = getClaimsFromAccessToken(session.accessToken);
 
-  return {
-    sessionId,
-    user: session.user,
-    organizationId,
-    role,
-    permissions,
-    impersonator: session.impersonator,
-    accessToken: session.accessToken,
-  };
+  return json(
+    {
+      sessionId,
+      user: session.user,
+      organizationId,
+      role,
+      permissions,
+      impersonator: session.impersonator,
+      accessToken: session.accessToken,
+      ...data,
+    },
+    {
+      headers: {
+        ...headers,
+        ...session.headers,
+      },
+    },
+  );
 }
 
 async function terminateSession(request: Request) {
-  const { sessionId } = await withAuth(request);
+  const encryptedSession = await getSession(request.headers.get('Cookie'));
+  const { accessToken } = (await getSessionFromCookie(request.headers.get('Cookie'))) as Session;
```

Comment:
Can we just pass the encrypted session directly to `getSessionFromCookie` and make that sync? Prevents a little extra work.